Implement handling of the llvm.fabs intrinsic

futile opened the pull request:
This PR adds support for the `llvm.fabs` intrinsic, and adds a test.

The test fails with current KLEE during intrinsic lowering, as LLVM can not/does not lower `llvm.fabs` further.

To change this, we ignore calls to the `llvm.fabs` intrinsic during intrinsic lowering, and instead handle them during `Executor::executeCall`, where the implementation is straight-forward, using `llvm::abs()`.

I moved up the definition of `fpWidthToSemantics` as it now needs to be called earlier.

Note: A call to `llvm.fabs` usually results from using `math.h`'s `fabs()` (it does so for my clang 8.0). In order to not rely on the behavior of my compiler, I added a `.ll` file instead of a `.c` file (although it was generated from one) for the test.

ccadar commented:
Thanks for this contribution, @futile !  This looks great, may I just ask for two small things:
1) Can you run `clang-format` -- I spotted a missing space after an `if`
2) Can you add the original C test file as a comment, so that we can regenerate it later if needed?

futile commented:
Thanks for the review @ccadar ! 
1. I ran `clang-format` on the regions I changed (running it on the whole `Executor.cpp` resulted in a suspiciously large diff). That fixed the missing space, but also introduced some linebreaks etc.
2. The original C test file is now a comment in the `.ll` file, together with instructions on how it was generated and amended for older LLVM versions.

ccadar commented:
@futile Excellent, that's what I had in mind!
